_docs/how-to/python-package.md

```diff
@@ -0,0 +1,207 @@
+and is not intended for external users.
+
+(python-package-distribution-pypi)=
+### PyPI
+
+Use [trusted publishing](https://docs.pypi.org/trusted-publishers/) to publish directly from your github repository.
+You can use the [actions/attest-build-provenance](https://github.com/actions/attest-build-provenance)
+and [pypa/gh-action-pypi-publish](https://github.com/pypa/gh-action-pypi-publish) actions in your workflows.
+Trigger the workflow based on a version tag being pushed.
+Feel free to check out this project's workflows for an example.
+Make sure that your repository only allows trusted contributors write access.
+
+The team manager and another truster team member should be the package owners on PyPI,
+using their Canonical email addresses.
+Make sure to also claim your package on [Test PyPI](https://test.pypi.org/),
+and setup a workflow for publishing there.
+All team members can be owners on Test PyPI.
+
+A major benefit of publishing on PyPI is that users of your library can specify version ranges in their dependencies.
+Therefore, if you’re going to publish on PyPI, we highly recommend that you use semantic versioning for your library.
+
+A non-dev/alpha/beta/etc qualified 1.x release to PyPI signifies that your library is ready for public consumption.
+You should also communicate this through the ["Development Status" Trove classifier](https://pypi.org/classifiers/) in your `pyproject.toml`.
+
+(python-package-distribution-git)=
+### Git
+
+You can get started by distributing your library as a Python package with very little friction using GitHub.
+This is good for prototyping, or when first transitioning from a charm-lib to a Python package,
+and may be a good fit for libraries that are intended for team-internal use.
+Once you’re done prototyping and the library is ready for external users, you’ll want to start using PyPI instead.
+
+Git dependencies are limited in that they don’t allow for sophisticated dependency resolution.
+You can only specify an exact reference (tag, commit, or branch).
+You can’t specify a version range.
+This is problematic if your library has dependencies,
+as having to request a specific version of your library makes it more likely that any dependency clashes will require manual intervention.
+It becomes even more problematic if your library may be depended on by other charm libraries.
+Requesting an exact hash or a branch tip may be sufficient for team-internal projects and prototyping,
+but proper versioning support is critical when sharing your library more widely,
+which will require promoting your package to PyPI.
+Tools that scan for security vulnerabilities may also struggle with such dependencies.
+
+You’ll need to include git in your charm’s build dependencies to use a GitHub-hosted library in your charm:
+
+```yaml
+parts:
+  charm:
+    build-packages: [git]
+```
+
+Then you can specify the dependency in your requirements like this:
+
+```
+ops @ git+https://github.com/canonical/operator@main
```

Comment:
Would it be better to present a `charmlibs` example using the naming suggested above? Using `ops` here seems a bit confusing in this context.